### ISSUE TITLE
0.8.0-rc9+1 - the node is receiving the same block with different timestamps

### ISSUE THREAD

**Issue by dorin100** [input-output-hk/jormungandr#1303]
**Describe the bug**
My local node is shown as synced but it is receiving the same block with different timestamps.

**Mandatory Information**
```
jcli 0.8.0-rc9+1 (HEAD-3339987e, release, windows [x86_64]) - [rustc 1.39.0 (4560ea788 2019-11-04)]
jormungandr 0.8.0-rc9+1 (HEAD-3339987e, release, windows [x86_64]) - [rustc 1.39.0 (4560ea788 2019-11-04)]
```

**To Reproduce**
1. start local node, connected to qa testnet
2. wait for a couple of epochs - everything is ok
3. stop the local node for a couple of epochs
4. start the local node --> it is shown as synced but it receives the same block hash/date with a different timestamp

**Expected behavior**
The node should receive the latest tip from the network.
If the node is quarantined, it should be displayed so and the user should now what to do in order for the node to sync correctly.

**Additional context**
- OS: win 10
- testnet: QA
- node logs:
```
Dec 08 12:58:04.157 ERRO cannot process network headers, reason: deadline has elapsed, task: block
Dec 08 12:58:04.247 ERRO cannot process network headers, reason: deadline has elapsed, task: block
Dec 08 12:58:04.297 ERRO cannot process network headers, reason: deadline has elapsed, task: block
Dec 08 12:58:04.344 INFO response stream failed, reason: the operation was aborted, request: PullHeaders, node_id: 49e171f81e3abb62bb3360946ba67a2a05129ef9b9a84030, peer_addr: 3.123.160.108:3000, task: network
Dec 08 12:58:04.360 INFO response stream failed, reason: the operation was aborted, request: PullHeaders, node_id: 02c3c4398064548430bad21718599d10cb457d82a1b32ff5, peer_addr: 3.120.96.219:3000, task: network
Dec 08 12:58:04.370 INFO response stream failed, reason: the operation was aborted, request: PullHeaders, node_id: a85bfdff5bd4ea55beac7d5644e71958db6cb2c6ec600553, peer_addr: 54.193.75.55:3000, task: network
Dec 08 12:58:04.576 ERRO cannot process network headers, reason: deadline has elapsed, task: block
Dec 08 12:58:04.584 INFO response stream failed, reason: the operation was aborted, request: PullHeaders, node_id: 4d2cb5f9a4eec68dead6b45f393c942e838e96c948757b31, peer_addr: 3.124.45.245:3000, task: network
Dec 08 12:58:04.959 INFO response stream failed, reason: the operation was aborted, request: PullHeaders, node_id: a8f4d660ff8fb38d72fb4b521a869e317a278bd5e55b68b0, peer_addr: 3.124.201.85:3000, task: network
```
[node_logs_info.txt](https://github.com/input-output-hk/jormungandr/files/3935911/node_logs_info.txt)
[node_logs_debug.zip](https://github.com/input-output-hk/jormungandr/files/3935912/node_logs_debug.zip)

- node stats
```
$ jcli rest v0 node stats get --host "http://127.0.0.1:9000/api"
---
blockRecvCnt: 680
lastBlockDate: "36.317"
lastBlockFees: 0
lastBlockHash: 1cc208efd431f292aa1c2dd531cc663560ece0f5502c2fb4a5ac46a8dccd9f88
lastBlockHeight: "5326"
lastBlockSum: 0
lastBlockTime: "2019-12-08T05:52:25+00:00"
lastBlockTx: 0
state: Running
txRecvCnt: 0
uptime: 133
version: jormungandr 0.8.0-rc9+1-3339987e
```
```
$ jcli rest v0 node stats get --host "http://127.0.0.1:9000/api"
---
blockRecvCnt: 796
lastBlockDate: "36.317"
lastBlockFees: 0
lastBlockHash: 1cc208efd431f292aa1c2dd531cc663560ece0f5502c2fb4a5ac46a8dccd9f88
lastBlockHeight: "5326"
lastBlockSum: 0
lastBlockTime: "2019-12-08T06:10:40+00:00"
lastBlockTx: 0
state: Running
txRecvCnt: 0
uptime: 1240
version: jormungandr 0.8.0-rc9+1-3339987e
```
```
$ jcli rest v0 node stats get --host "http://127.0.0.1:9000/api"
---
blockRecvCnt: 842
lastBlockDate: "36.317"
lastBlockFees: 0
lastBlockHash: 1cc208efd431f292aa1c2dd531cc663560ece0f5502c2fb4a5ac46a8dccd9f88
lastBlockHeight: "5326"
lastBlockSum: 0
lastBlockTime: "2019-12-08T06:19:00+00:00"
lastBlockTx: 0
state: Running
txRecvCnt: 0
uptime: 1739
version: jormungandr 0.8.0-rc9+1-3339987e
```

**Comment by dorin100** [comment]
restarting the node a couple of times fixed the issue.

**Comment by Exteroqs** [comment]
i have the similar problem en the 0.8.0
```
ERRO cannot process network headers, reason : deadline has elapsed, task: network
WARM response canceled by the processing task, request: PushHeaders, sub_task: server, task: network
```
These ads appeared interspersed, within the typical ones that appear, sometimes they only appeared "warm" , and yes, restarting the node a couple of times resolved the problem.

**Comment by mzabaluev** [comment]
There is #1349 to reduce redundant requests.

**Comment by mzabaluev** [comment]
Should have been fixed at the same time with #1402.
